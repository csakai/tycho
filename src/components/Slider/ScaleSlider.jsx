import React from 'react';
import Slider from './Slider';

export default class ScaleSlider extends React.Component {
  render() {
    return (
      <div>
        <div className="slider slider--horizontal">
          <span className="slider__label">
            <span>{this.props.label}: &times; 10</span>
            <sup>{this.props.value}</sup>
          </span>
          <Slider
            orientation="horizontal"
            value={this.props.value}
            onChange={this.props.onChange}
            min={0}
            max={10}
          />
        </div>
      </div>
    );
  }
}
