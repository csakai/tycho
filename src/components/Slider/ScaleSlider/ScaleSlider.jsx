import React from 'react';
import Slider from '../Slider';
import PropTypes from 'prop-types';

export default class ScaleSlider extends React.Component {

    static propTypes = {
        label: PropTypes.string,
        value: PropTypes.number,
        onChange: PropTypes.func,
        min: PropTypes.number,
        max: PropTypes.number
    }

    render() {
        return (
            <div>
                <div className="slider slider--horizontal">
                    <span className="slider__label">
                        {this.props.label}
                        {this.props.value || this.props.min}
                    </span>
                    <Slider
                        orientation="horizontal"
                        value={this.props.value}
                        onChange={this.props.onChange}
                        defaultValue={this.props.min}
                        min={this.props.min}
                        max={this.props.max}
                    />
                </div>
            </div>
        );
    }
}
