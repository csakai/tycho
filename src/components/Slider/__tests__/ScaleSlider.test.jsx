import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import ScaleSlider from '../ScaleSlider';

describe('Scale Slider Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<ScaleSlider
      value={40}
      label="Test Slider"
      onChange={jest.fn()}
    />);
  });

  describe('render()', () => {
    it('should successfully render the slider', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
