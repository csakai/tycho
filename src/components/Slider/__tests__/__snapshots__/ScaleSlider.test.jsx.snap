// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scale Slider Component render() should successfully render the slider 1`] = `
<div>
  <div
    className="slider slider--horizontal"
  >
    <span
      className="slider__label"
    >
      <span>
        Test Slider
        : Ã— 10
      </span>
      <sup>
        40
      </sup>
    </span>
    <Slider
      max={10}
      min={0}
      onChange={[Function]}
      orientation="horizontal"
      value={40}
    />
  </div>
</div>
`;
